compareBuckets/compareBuckets.js

```diff
@@ -0,0 +1,140 @@
+                            dstMarker = marker;
+                            statusObj.dstKeyMarker = dstMarker;
+                            return _done();
+                        });
+                },
+            }, err => {
+                if (err) {
+                    return done(err);
+                }
+
+                let srcIdx = 0;
+                let dstIdx = 0;
+
+                if (srcDone && srcContents.length === 0) {
+                    while (dstIdx < dstContents.length) {
+                        log.error('missing object in source', {
+                            bucket: bucketdSrcParams.bucket,
+                            object: dstContents[dstIdx].key,
+                            objectUrl: getObjectURL(bucketdDstParams.bucket, dstContents[dstIdx].key),
+                        });
+                        ++statusObj.missingInSrcCount;
+                        ++dstIdx;
+                    }
+                }
+
+                if (dstDone && dstContents.length === 0) {
+                    while (srcIdx < srcContents.length) {
+                        log.error('missing object in destination', {
+                            bucket: bucketdDstParams.bucket,
+                            object: srcContents[srcIdx].key,
+                            objectUrl: getObjectURL(bucketdSrcParams.bucket, srcContents[srcIdx].key),
+                        });
+                        ++statusObj.missingInDstCount;
+                        ++srcIdx;
+                    }
+                }
+
+                while (srcIdx < srcContents.length && dstIdx < dstContents.length) {
+                    if (srcContents[srcIdx].key < dstContents[dstIdx].key) {
+                        log.error('missing object in destination', {
+                            bucket: bucketdDstParams.bucket,
+                            object: srcContents[srcIdx].key,
+                            objectUrl: getObjectURL(bucketdSrcParams.bucket, srcContents[srcIdx].key),
+                        });
+                        ++statusObj.missingInDstCount;
+                        ++srcIdx;
+                        continue;
+                    }
+
+                    if (srcContents[srcIdx].key > dstContents[dstIdx].key) {
+                        log.error('missing object in source', {
+                            bucket: bucketdSrcParams.bucket,
+                            object: dstContents[dstIdx].key,
+                            objectUrl: getObjectURL(bucketdDstParams.bucket, dstContents[dstIdx].key),
+                        });
+                        ++statusObj.missingInSrcCount;
+                        ++dstIdx;
+                        continue;
+                    }
+
+                    ++dstIdx;
+                    ++srcIdx;
+                }
+
+                srcContents = srcContents.slice(srcIdx);
+                dstContents = dstContents.slice(dstIdx);
+                return setTimeout(() => done(null), 500);
```

Comment:
Is there a reason for the `setTimeout`? I think it could make things painfully slow on a powerful platform.

Reply:
this was a precaution against making rapid requests. I can remove it if this presents itself as a bottleneck